plugins/sdex.go

```diff
@@ -813,3 +813,13 @@ func (sdex *SDEX) GetLatestTradeCursor() (interface{}, error) {
 
 	return records[0].PT, nil
 }
+
+// GetOrderBook gets the SDEX order book
+func GetOrderBook(api *horizon.Client, assetBase *horizon.Asset, assetQuote *horizon.Asset) (orderBook horizon.OrderBookSummary, e error) {
```

Comment:
This should implement the method signature from `TradeAPI` in `exchange.go` as closely as possible:

    GetOrderBook(pair *model.TradingPair, maxCount int32) (*model.OrderBook, error)

sdex.go holds a `tradingPair` object, we should just use that when creating the OrderBook and any related Orders. For now, we should also return an error if the passed in `pair` is different than the `tradingPair` saved.

---

also as a general kelp convention we can leave the return values in the signatures unnamed -- the convention is to only name them if they are of the same type so it's a better API. here they are different types so we can leave them unnamed

Reply:
I wrote it this way specifically to allow for non-matching feeds; do you want to disallow using another asset's feed for now?

Reply:
ah, I see. that makes sense.

In that case I'd recommend creating a new sdex instance in the `case` statement of priceFeed.go (no global var needed as commented above). Here's a list of the all the inputs to the factory method, we can stub out those values that are not needed for this call:

    func MakeSDEX(
        api *horizon.Client,    // this can be a global var for now _sdexHack.api
        sourceSeed string,    // can be ""
        tradingSeed string,    // can be ""
        sourceAccount string,    // can be ""
        tradingAccount string,    // can be ""
        network build.Network,    // this can be a global var for now _sdexHack.network
        threadTracker *multithreading.ThreadTracker,    // this can be nil
        operationalBuffer float64,    // can be 0
        operationalBufferNonNativePct float64,    // can be 0
        simMode bool,    // can be true
        pair *model.TradingPair,    // should be a valid pair, see trade.go for how we make this
        assetMap map[model.Asset]horizon.Asset,    // should be a valid map for the two assets, see trade.go for how we construct this
    )

Reply:
we should be able to change this function signature now to:

    GetOrderBook(pair *model.TradingPair, maxCount int32) (*model.OrderBook, error)

where we can use the tradingPair in sdex.go where needed

Reply:
done